app/Browsershot/BrowsershotService.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Browsershot;
+
+use App\Browsershot\Modes\Pdf as PdfFactory;
+use App\Browsershot\Modes\Screenshot as ScreenshotFactory;
+use App\Contracts\FileInterface;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use Spatie\Browsershot\Browsershot;
+
+class BrowsershotService
+{
+    public const TYPE_SCREENSHOT = 'screenshot';
+    public const TYPE_PDF = 'pdf';
+
+    public function __construct(
+        protected Browsershot $browsershot,
+        protected PdfFactory $pdfFactory,
+        protected ScreenshotFactory $screenshotFactory
+    ) {}
+
+    public function execute(string $type, string $input, Collection $parameters): ?FileInterface
+    {
+        $type = Str::lower($type);
+
+        return match ($type) {
+            self::TYPE_SCREENSHOT => $this->screenshot($input, $parameters),
+            self::TYPE_PDF => $this->pdf($input, $parameters),
+            default => throw new \RuntimeException('Type method "' . $type . '" not implemented'),
+        };
+    }
+
+    protected function screenshot(string $url, Collection $parameters): FileInterface
+    {
+        $factory = $this->screenshotFactory;
+        $factory->setUrl($url);
+
+        $factory->setSize(
```

Comment:
You are using strict types, but all these parameters are not casted. This may cause unnecessary type errors. You should increase the phpstan level so that these kind of errors become visible.